internal/kubernetes/kubernetes.go

```diff
@@ -529,7 +529,7 @@ func CA() ([]byte, error) {
 }
 
 // Find the first suitable Service, filtering on infrahq.com/component
-func (k *Kubernetes) Service(component string) (*corev1.Service, error) {
+func (k *Kubernetes) Service(component string, labels ...string) (*corev1.Service, error) {
```

Comment:
It seems really surprising to me that this label arg will be filtered and only the label matching `app.kubernetes.io/instance` will be used.

Maybe that filtering should also be done by the caller?

At the very least we should document how labels are used in the godoc above here.

Reply:
Good point, switched the filtering to be done by the callers